src/imperative.rkt

```diff
@@ -98,142 +98,157 @@
             (if (equal? (language-name (*lang*)) "sollya") ; Sollya has slightly different if
-                (printf "~aif ~a then {\n" indent 
-                        (format-cond (trim-infix-parens 
-                                        (convert-expr cond #:ctx ctx #:indent indent))))
-                (printf "~a~a\n~aif ~a {\n" indent (convert-declaration ctx outvar)
-                        indent (format-cond (trim-infix-parens 
-                                              (convert-expr cond #:ctx ctx #:indent indent)))))
-            (printf "~a\t~a\n" indent
-                    (convert-assignment outvar (convert-expr ift #:ctx ctx
-                                        #:indent (format "~a\t" indent))))]
+                (printf "~aif ~a then {\n" indent (format-cond (trim-infix-parens cond*)))
+                (printf "~a~a\n~aif ~a {\n" indent (convert-declaration ctx* outvar)
+                        indent (format-cond (trim-infix-parens cond*))))
+            (printf "~a\t~a\n" indent (convert-assignment outvar ift*))
+            (loop (cdr branches) #f (cons outvar ift-prec))]
            [(list _ #t else) ; branch condition #t => else branch
-            (printf "~a} else {\n" indent)
-            (printf "~a\t~a\n" indent
-                    (convert-assignment outvar (convert-expr else #:ctx ctx
-                                        #:indent (format "~a\t" indent))))
-            (printf "~a}~a\n" indent (if (equal? (language-name (*lang*)) "sollya") ";" ""))]
+            (define-values (else* else-prec) (convert-expr else #:ctx ctx #:indent (format "~a\t" indent)))
+            (printf "~a} else {\n~a\t~a\n" indent indent (convert-assignment (car out) else*))
+            (printf "~a}~a\n" indent (if (equal? (language-name (*lang*)) "sollya") ";" ""))
+            (values (car out) (cdr out))]
            [(list _ cond elif) ; otherwise => else if branch
+            (define-values (cond* cond-prec) (convert-expr cond #:ctx ctx #:indent indent))
+            (define-values (elif* elif-prec) (convert-expr elif #:ctx ctx #:indent (format "~a\t" indent)))
             (printf
               (if (equal? (language-name (*lang*)) "sollya") ; Sollya has slightly different else if
                   "~a} else if ~a then {\n"
                   "~a} else if ~a {\n")
-              indent
-              (format-cond (trim-infix-parens (convert-expr cond #:ctx ctx #:indent indent))))
-            (printf "~a\t~a\n" indent
-                    (convert-assignment outvar
-                                        (convert-expr elif #:ctx ctx
-                                                      #:indent (format "~a\t" indent))))])))
-      outvar]
+              indent (format-cond (trim-infix-parens cond*)))
+            (printf "~a\t~a\n" indent (convert-assignment (car out) elif*))
+            (loop (cdr branches) first? out)])))]
```

Comment:
`if` is broken. Consider the following example:
```
(FPCore if (x)
  (if TRUE x (let ([y (/ 1 0)]) y)))
```
The translated result is:
```c
double ex6(double x) {
	double tmp;
	if ((double) TRUE) {
		tmp = x;
		double y = 1.0 / 0.0;
	} else {
		tmp = y;
	}
	return tmp;
}
```
Maybe it is possible to fix this issue by changing `current-output-port` for `ift` and `elif` expressions.

Reply:
Thanks. I just messed up the order of the `printf`'s.